### PR TITLE
MoreThen5000Records

### PR DESCRIPTION
**Pull request by iggsn**
Did some cleanup and added ability to export more than 5000 records.
Also updated to latest libraries.

### CODE REVIEW

File: AuditHistoryExtractor/MyPluginControl.cs [hunk 1]
```diff
@@ -427,29 +417,41 @@ private void RetrieveAuditHistory(RetrieveAuditHistoryMode retrieveMode)
...
-                            ev.Result = MessageToMuchRecords;
-
-                            return;
-                        }
-
-
+                            var xml = FetchXMLHelper.AddCookie(newFetch, cookie, page);
+                            EntityCollection collection = Service.RetrieveMultiple(new FetchExpression(xml));
+
+                            if (recordsExtracted != null && !collection.EntityName.Equals(currentEntitySelected))
```
Comment (andreasziegl) [inline AuditHistoryExtractor/MyPluginControl.cs]: The recordsExtracted != null does not serve a reason or?

**Comment by andreasziegl** [comment]
Very nice and usefull change

File: AuditHistoryExtractor/MyPluginControl.cs (hunk 1 above)
Comment (iggsn) [reply AuditHistoryExtractor/MyPluginControl.cs]: I think you are right. I haven't changed all the code. This lines probably can be removed. I fear more, that this project is not maintained anymore and that change will not make it into a new release. :-(

**Comment by andreasziegl** [comment]
Hm yeah I see what you mean. I also have that feeling. Let's see if @alduzzen1985 has still interest in maintaining this awesome plugin.
If not then we can see in the next year what we can do about that. Maybe the guys from MscrmTools have a standard procedure.
Would be really a pity if this tool gets lost

**Comment by iggsn** [comment]
We will find a way to bring it back to live.

**Comment by alduzzen1985** [comment]
Hi @iggsn @andreasziegl , thanks a million for the Fix. I'll make sure to deploy in these days your fix. Sorry for the delay but I had crazy days.

File: AuditHistoryExtractor/MyPluginControl.cs (hunk 1 above)
Comment (iggsn) [reply AuditHistoryExtractor/MyPluginControl.cs]: This need to be changed to !List.Any() to make sure, we retrieved anything from the fetch.

**Pull request merged**